unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarModifierAreaHUD/Scripts/AvatarModifierAreaFeedbackView.cs

```diff
@@ -0,0 +1,233 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using Cysharp.Threading.Tasks;
+using TMPro;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace DCL.AvatarModifierAreaFeedback
+{
+    public class AvatarModifierAreaFeedbackView : MonoBehaviour, IAvatarModifierAreaFeedbackView, IPointerEnterHandler, IPointerExitHandler
+    {
+        
+        internal enum AvatarModifierAreaFeedbackState { NEVER_SHOWN, ICON_VISIBLE, WARNING_MESSAGE_VISIBLE }
+
+        private const string PATH = "_AvatarModifierAreaFeedbackHUD";
+        private const string PATH_TO_WARNING_MESSAGE = "_WarningMessageAreaFeedbackHUD";
+        private BaseRefCounter<AvatarAreaWarningID> avatarAreaWarningsCounter;
+        
+        [SerializeField] private CanvasGroup warningMessageCanvasGroup;
+        [SerializeField] private CanvasGroup warningIconCanvasGroup;
+        [SerializeField] private RectTransform messageContainer;
+        internal float animationDuration;
+        internal bool isVisible;
+        internal AvatarModifierAreaFeedbackState currentState;
+        internal Dictionary<AvatarAreaWarningID, GameObject> warningMessagesDictionary;
+        internal CancellationTokenSource deactivatePreviewCancellationToken = new CancellationTokenSource();
+        internal CancellationTokenSource deactivateIconAnimationToken = new CancellationTokenSource();
+        internal CancellationTokenSource deactivateWarningMesageAnimationToken = new CancellationTokenSource();
```

Comment:
I gave it a look; it's possible to combine everything into one UniTask animation. 

That would mean adding a different condition checker depending on the state of the message; that takes into account the state of the message and the action that fired the state change (`OnPointerExit`, `OnPointerEnter`, or entering/leaving the area) through the parameters. Having it in a separate method allows calling them independently. 

Therefore, yes, I could do it. But I think it's more flexible to leave it as it is, to avoid a big animation that has all the possibilities in its parameters.

Finally, the animation is not yet set to stone; it's going through analysis with Romi and the animator. So leaving it like this may make it more flexible to change, which will most probably do.

Reply:
Fair enough!